### DISCUSSION TITLE
Integration with hosted payment pages

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1268] [Q&A]
Any sample implementation in this regard?

**Comment by michaelbromley** [comment]
Hi, you can take a look at our upcoming [Mollie plugin](https://github.com/vendure-ecommerce/vendure/blob/minor/packages/payments-plugin/src/mollie/mollie.plugin.ts), which uses a hosted pages with redirect flow.

**Reply by ramkishore-yerubandi** [reply]
Thanks for the guidance, I was able to implement this for PayU gateway. There is a followup question... if the payment fails or throws an error... how do I transition the payment entry to "Error" | "Declined" with errorMessage from controller of the plugin

**Reply by michaelbromley** [reply]
You could use [OrderService.transitionPaymentToState()](https://www.vendure.io/docs/typescript-api/services/order-service/#transitionpaymenttostate).
